crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -203,15 +203,16 @@ fn try_into_byte(v: &Value) -> Option<u8> {
 #[serde_as]
 #[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
 pub enum EventFilter {
+    /// Query by events that match all the given filters, or return all events if no filters are
+    /// supplied.
+    All(Vec<EventFilter>),
```

Comment:
only empty vec is implemented, we can just do `All` here?  one step further, having an `All` filter seems un-necessary and we can change `query: EventFilter` to `Option<EventFilter>`? Both can be more breaking so okay to me if we want to minimize the breaking here.

Reply:
Yeah, I'm not too happy with how `All` works today and the fact that this is how you communicate "no filter" (vs. how it's done for transactions which is to omit the filter), but I opted to do it this way to minimise breakage.

Reply:
would it be clearer to have this be a fixed-size, 0-sized array? Assuming that it minimizes breakage in the same way as we have here?

Reply:
I initially tried to make this a unit type for similar reasons, but that made serde very unhappy. I can try `[EventFilter; 0]` though!

Reply:
0-size array worked! 🎉

Reply:
huzza for the type system!